### ISSUE TITLE
rbconfig and other built-in libraries don't load without stdlib

### ISSUE THREAD

**Issue by headius** [jruby/jruby#6362]
In #6340 I found that one of our tests tries to clear the `jruby.home` property and then `require "rbconfig"`. However we load the rbconfig extension via rbconfig.rb here: https://github.com/jruby/jruby/blob/master/lib/ruby/stdlib/rbconfig.rb

There may be other built-in libraries that depend on a single `load_ext` in stdlib.

We should consider moving such files into the base jruby.jar file, since they are frequently necessary for normal operation of Ruby, with or without a standard library present.

**Comment by kares** [comment]
good find, makes sense and I am surprised the tests did not fail before.
as this was [changed](https://github.com/jruby/jruby/commit/fb3ba4b3543b72e735ba53b51e6b7ec8a9916e1b#diff-a40568b9da02963919222c8b1eb53774L1752) due bootsnap (in 9.2.6.0) to always have explicit *library.rb* files around compared to "virtual" .rb ones.

**Comment by headius** [comment]
@kares Looks like you have this in hand, let me know if you want a review.

**Comment by kares** [comment]
@headius not really, have only been adjusting tests to not assume (previous) built-in lazy lib behavior at https://github.com/jruby/jruby/pull/6340

**Comment by headius** [comment]
@kares Ahh ok... so this is still an open task.

**Comment by kares** [comment]
for (explicit) reference, here's the commit changing the tested behavior: https://github.com/jruby/jruby/commit/a6b428ba47b647da4336353b8a754913277d9028

**Comment by headius** [comment]
On second thought this will drastically change in 9.4 so there's not much value in doing it now. Moving this and the PR to 9.4 to be reexamined once more libraries have moved out to gems.

**Comment by headius** [comment]
I did a bit of spelunking and the original reason we moved to having mostly-empty .rb files in stdlib was to address [issues integrating with load-caching tools like bootsnap](https://github.com/Shopify/bootsnap/issues/162): https://github.com/jruby/jruby/commit/fb3ba4b3543b72e735ba53b51e6b7ec8a9916e1b

I'm not sure if there's a resolution that removes the .rb files from the filesystem without breaking the expectation of files-on-filesystem. The workaround, for now, is just using jruby-complete (or the jruby-stdlib jar combined with jruby core jars) so the files in question can be found without being on the filesystem, but I would assume running in this fashion probably isn't compatible with bootsnap.

I might argue that changes are needed in bootsnap to handle "internal requires" like we used to do, but I don't think it's going to happen without some community help.

Closing this for now.
